service/s3/s3crypto/crypto.go

```diff
@@ -41,3 +42,11 @@ func (rc *CryptoReadCloser) Read(b []byte) (int, error) {
 	}
 	return rc.Decrypter.Read(b)
 }
+
+type bytesWriteSeeker struct {
+	bytes.Buffer
+}
+
+func (ws *bytesWriteSeeker) Seek(offset int64, whence int) (int64, error) {
```

Comment:
Would a `aws.WriteAtBuffer` or similar be good to use here instead of a bytes buffer? Implementing Seek, but disabling its functionality could cause hard to track down errors.  It works in the use case here, but if any other usages of seek are added such as seeking to get lenght, tracking down the bug will be pretty hard.